### PR TITLE
add uniroot function to python

### PR DESCRIPTION
**Pull request by catoverdrive**
this exposes `uniroot` in python. I added a new (python) AST node, `LambdaFunction` which basically mimics `LambdaClassMethod` but doesn't have a caller.

cc @konradjk

### CODE REVIEW

File: python/hail/expr/functions.py [hunk 1]
```diff
@@ -3756,3 +3756,42 @@ def liftover(x, dest_reference_genome, min_match=0.95):
...
+
+    Examples
+    --------
+
+    >>> hl.uniroot(lambda x: x - 1, -5, 5).value
+    1.0
+
+    Notes
+    -----
+    `f(min)` and `f(max)` must have opposite signs.
```
Comment (jbloom22) [inline python/hail/expr/functions.py]: Not true, they can be zero. more correct to say "must not have the same sign."

I don't see this condition checked anywhere, add it inside the scala code? In R, results in an error message:
```
Error in uniroot(f1, c(0, 10)) : 
  f() values at end points not of opposite sign
```

File: python/hail/expr/functions.py [hunk 2]
```diff
@@ -3756,3 +3756,42 @@ def liftover(x, dest_reference_genome, min_match=0.95):
...
+    ----------
+    f : function ( (arg) -> :class:`.Float64Expression`)
+        Must return a :class:`.Float64Expression`.
+    min : :class:`.Float64Expression`
+    max : :class:`.Float64Expression`
+
+    Returns
+    -------
+    :class:`.Expression`
+        Expression whose type is the element type of the collection.
```
[the remark this answers is not in the document]
Comment (catoverdrive) [reply python/hail/expr/functions.py]: haha

oops

:(

File: python/hail/expr/functions.py (hunk 1 above)
Comment (catoverdrive) [reply python/hail/expr/functions.py]: it's inside the scala code, yeah.

Comment (jbloom22) [reply python/hail/expr/functions.py]: Where?

**Pull request merged**